docs/sso.md

```diff
@@ -0,0 +1,87 @@
+# Single Sign-On
+
+The consumerfinance.gov codebase support single sign-on (SSO) with [Open ID Connect (OIDC) providers](https://openid.net/). We use Mozilla's [mozilla-django-oidc](https://mozilla-django-oidc.readthedocs.io/en/stable/) library for OIDC support.
+
+## Local implementation
+
+Our implementation of OIDC SSO uses a custom subclass of mozilla-django-oidc's `OIDCAuthenticationBackend`, found in `login.auth.CFPBOIDCAuthenticationBackend`, that supports additional OIDC claims processing. This enables us to map OIDC roles to Django groups and/or superuser permissioning, and lets us synchronize a user's name to the profile information in the OIDC provider.
+
+if the `OIDC_OP_ADMIN_ROLE` setting is defined, Django's `is_superuser` will be set for users who the OIDC provider reports as having this role identifier.
+
+Every user who is authorized in the SSO provider will be assigned to a "Wagtail Users" group. This enables all members to log into the Wagtail admin.
+
+This group is created with the `wagtailadmin.access_admin` permission once via data migration in the `login` app. If for any reason this group is removed, it will need to be recreated, otherwise new SSO users will not be able to login without an admin user assigning them the `wagtailadmin.access_admin` permission manually.
+
+## Testing OIDC single sign-on locally
+
+For development of mozilla-django-oidc, Mozilla [uses a test OIDC provider](https://github.com/mozilla/docker-test-mozilla-django-oidc), which is published to Docker Hub as `mozilla/oidc-testprovider:oidc_testprovider-latest`. This test provider is a Django project that uses [django-oidc-provider](https://django-oidc-provider.readthedocs.io/) as the OIDC provider application.
+
+We make use of this test provider to enable local testing and development related to SSO for consumerfinance.gov. The OIDC test provider is available in the `docker-compose.sso.yml` file.
+
+For consumerfinance.gov, our compose file adds another set of fixture data that adds `localhost:8000` as a client in addition to the test provider's built-in [`testrp` client that Mozilla uses for local development](https://github.com/mozilla/mozilla-django-oidc?tab=readme-ov-file#local-development).
+
+!!! note "Limitations"
+
+    Mozilla's test provider does not provide group/role claims, just user authentication. This means that, while it enables testing the general SSO flow locally, it does not support provisioning new users in Wagtail (they will not receive access to the Wagtail admin).
+
```

Comment:
```suggestion
```

Since you changed the logic to remove the check for the regular user role, does this limitation still exist? I created a new user in the test provider and when I logged in, a Wagtail user was created who was able to access the admin.

Reply:
Removed in 8543484a3